source/common/ssl/context_impl.cc

```diff
@@ -223,6 +223,15 @@ bool ContextImpl::verifySubjectAltName(X509* cert,
   return verified;
 }
 
+bool ContextImpl::uriMatch(const std::string& uriPattern, const char* uri) {
+  size_t pattern_len = uriPattern.length();
+  if (pattern_len > 1 && uriPattern.substr(pattern_len - 2) == "/*") {
+    return strncmp(uriPattern.c_str(), uri, pattern_len - 1) == 0 && uri[pattern_len - 1];
+  }
+
+  return uriPattern == uri;
+}
+
 bool ContextImpl::dNSNameMatch(const std::string& dNSName, const char* pattern) {
```

Comment:
Sorry to bring this up ... I feel like how this function is called (line 203 in context_impl.cc) is not right. It should be:
dNSNameMatch(dns_name, config_san)
config_san should be the pattern, not dns_name from the cert.

It makes sense to me that we support wildcard in config, not in certificates. Unfortunately I didn't notice this when I modified this function. This was unnoticed because we don't have integration tests, nor did we tell users this wildcard is supported.
@mattklein123 Could you confirm?